### ISSUE TITLE
How can I get Account_id/User_id against user Email?

### ISSUE THREAD

**Issue by Zuhayer** [Armor-Payments/armor-payments-php#6]
I want shopper to make a payment to the merchant in the App I am developing.

Flow:
My clients app works with a shopper ordering an item for a traveler to purchase. If they accept each others terms, Shopper deposits the money in the 'merchant' account and the merchant subtracts his share and then pays the traveler after successful exchange of item.

Requirement:
Best practice to get Users Account_id/User_id from mobile App to php server side for transaction?

Should i ask the mobile dev team to show a web-view, user logs in and then they send me the ID or is there an api available by which I can check user's account existence using his email?

**Comment by 7krasov** [comment]
Hey, guys, any updates? Can anybody tell us is there any way to send payment request to existing user/account, which has been already registered (not created from our api requests)?

**Comment by mrmarkfrench** [comment]
The short version here is that it doesn't matter if the user already has an existing account (not created through your API requests). You just call the `accounts` resource and create a new account for them. If no account exists, it will be created. If they already have an account, we'll return the ID of the existing account.

The only way this will return an error about an account already existing is if they already have an account AND that account is already associated with your API key. So long as you record the account ID when you call the `accounts` resource the first time, and then don't try to call it again for the same user, you won't have a problem.

**Comment by 7krasov** [comment]
Thanks a lot, Mark! Now it is clear for me

**Comment by 7krasov** [comment]
One more question.

I have registered seller@account.com and buyer@account.com users on sandbox.armorpayments.com (manually), emails are just example in this comments. Both users have Payee settings as bank account set on Account Settings page.
When the user of my system buyer@account.com clicks pay button on my site, my script searches an accountId identifier in database for this user and if there is no data, creates new one as you adviced above.

        $buyerAccount = $this->getClient()->accounts()->create(array(
            'company'    => 'User company '.uniqid($user->id.'-'),
            'address'    => $user->address ?: 'User adress '.uniqid($user->id.'-'),
            'city'       => $user->city ?: 'User city '.uniqid($user->id.'-'),
            'state'      => $user->state ?: 'User state '.uniqid($user->id.'-'),
            'country'    => $user->country ?: 'User country '.uniqid($user->id.'-'),
            'zip'        => $user->zip ?: 'User zip '.uniqid($user->id.'-'),
            'user_name'  => $user->getFullName(),
            'user_email' => $user->email,
            'user_phone' => $user->phone ?: '+1 8005551234',
        ));
When I try to create an order 

        $orderData = [
            'order_type'  => 3, // An escrow order for services
            'seller_id'   => $sellerUser->user_id,
            'buyer_id'    => $buyerUser->user_id,
            'amount'      => $this->getFormattedAmount($quote->total),
            'summary'     => config('app.name').' services order '.$this->getPaymentIdentifier(),
            'description' => config('app.name').' services order '.$this->getPaymentIdentifier(),
            'milestones'  => $items,
            'invoice_num'        => $this->getPaymentIdentifier(),
            'purchase_order_num' => $this->getPaymentIdentifier(),
        ];

I get an error message in response:

    [status] => error
    [errors] => stdClass Object
        (
            [Partner] => Array
                (
                    [0] => Partner cannot be buyer or seller on order for which they are a partner.
                )

        )

Also I see the buyer@account.com user on my seller's Partners page in list on sandbox.armorpayments.com after creating account and user through php code.

Sorry for offtop but I think it is continue of your explanation how to store buyers in your system correctly

**Comment by 7krasov** [comment]
Please explain how to create an order

**Comment by mrmarkfrench** [comment]
The issue you were having was that your partner account is not allowed to be the buyer or seller.

One of the two users (buyer@account.com or seller@account.com) is the user who owns your API key. Our system does not allow the partner account to participate in the order as buyer or seller.

You should create a third ID to use so that you can have a partner, a buyer, and a seller, and all three are different accounts.

**Comment by 7krasov** [comment]
How it has happened that two different test accounts become partners? 
>> Also I see the buyer@account.com user on my seller's Partners page in list on sandbox.armorpayments.com after creating account and user through php code.

It has appeared only after php code executing above. Could you advice how to attach existing buyer account to seller using php code?

**Comment by 7krasov** [comment]
>> Could you advice how to attach existing buyer account to seller using php code?

I mean without become partners option

**Comment by mrmarkfrench** [comment]
The only way an account becomes a partner is for it to request an API key. There is no API call which can make an account a partner.

I don't think the problem is that you have two partners (though maybe you do?). I think it's more likely that one of the two accounts you're using is your partner (either the buyer or the seller).

You should have a total of 3 accounts. One partner, one buyer, one seller.

**Comment by 7krasov** [comment]
Please explain how to create:
- seller account
- buyer account
- partner account (I even don't understand what is it)

Because I already have about 5 different accounts and can't create order.
I want seller@account.com to create order for buyer@account.com programmatically
What preferences for these 3 accounts should be in
- https://sandbox.armorpayments.com user panel?
- php code while creating accounts or smth else?

**Comment by mrmarkfrench** [comment]
The partner account is the account that has the API key. It looks like you already have an API key, so this part is already done.

The buyer and seller accounts are both created using the code you showed above in your example of creating the buyer account.

Your code is correct, the problem is that one of the accounts you're trying to use as either the buyer or seller is the one that owns the API key.

**Comment by 7krasov** [comment]
Now I see :-) I've made an order. Thank you!
